scripts/post-install/post-install.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const Logger = require('../utils/logger');
+
+/**
+ * Validates the installation of terra-toolkit.
+ * Logs a warning if terra-toolkit is detected as a hard dependency.
+ * terra-toolkit is expected to always be installed as a devDependency.
+ */
+function validateInstall() {
+  /**
+   * INIT_CWD returns the initial working directory of the npm install.
+   * See the following issue for more information: https://github.com/npm/npm/issues/16990
+   * Note: This feature was introduced in npm 5.4.0.
+   */
+  if (!process.env.INIT_CWD) {
+    Logger.warn('WARNING: [terra-toolkit] - INIT_CWD was undefined. This is likely due to an outdated version of npm. Please consider upgrading.');
+    return;
+  }
+
+  fs.readFile(`${process.env.INIT_CWD}/package.json`, (error, file) => {
+    if (error) {
+      Logger.warn('WARNING: [terra-toolkit] - Unable to read package.json in post-install script.');
```

Comment:
```suggestion
      Logger.warn('WARNING: [terra-toolkit] - Unable to validate if terra-toolkit was defined as a dev-dependency. Please ensure terra-toolkit has only added as a dependency. .');
```

Reply:
I think there were a few mistakes in the suggestion, I added something very similar: 9ee6b3d

WARNING: [terra-toolkit] - Unable to validate if terra-toolkit was installed as a dev-dependency. Please ensure terra-toolkit is defined as a devDependency and not a dependency in the package.json

Reply:
lol yeah my description wasn't very accurate